konnectivity-client/pkg/client/client.go

```diff
@@ -161,7 +178,7 @@ func (t *grpcTunnel) Dial(protocol, address string) (net.Conn, error) {
 	}
 
 	random := rand.Int63() /* #nosec G404 */
-	resCh := make(chan dialResult)
+	resCh := make(chan dialResult, 1)
```

Comment:
For the close case, it would be good to do this fix (buffer 1) on the closeCh on line 219.